framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2842,6 +2852,86 @@ public AnnotatedDeclaredType fromNewClass(NewClassTree newClassTree) {
         return type;
     }
 
+    /**
```

Comment:
Is there a reason why you moved this here from `AnnotatedTypes`? Do you expect subclasses might want to override this method?